### PR TITLE
Update description of flow.device.mode

### PR DESCRIPTION
**Pull request by PrasenjitAdhikary**
At present, OTG Flow configuration with multiple <b>device endpoints</b> is not supported in Ixia-C implementation. Primary reason, if the Tx/Rx device EP count > 1, the behavior of meshing mode and how it would impact the generated packets in the wire are not very clear. Users can easily create a wrong configuration based on an incorrect assumption, leading to a mismatch with the expected test result. And we need to support multiple Tx/Rx device EP, so that RSVP traffic with multiple LSP can be packed into one flow configuration, to achieve a target LSP scale of 64K.

With the proposed description change, it will be easy to correlate meshing mode with auto packet fields,  and understand how auto fields will be translated to packets in the wire, so that non-auto fields can be configured properly.

[View the proposed changes using the redocly API viewer](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/open-traffic-generator/models/update_mode_description/artifacts/openapi.yaml&nocors#tag/Configuration/operation/set_config)

A detailed document with use cases & examples is available [here](https://keysighttech.sharepoint.com/:w:/r/sites/Athena-G/Shared%20Documents/Architecture/RSVP/RSVP-DataTraffic/TrafficSupportMultipleEPsVer2.docx?d=wbe1f4dd553bb4df5bf944ffc962b2b13&csf=1&web=1&e=26AENh)

### CODE REVIEW

**Comment by bortok** [comment]
The description is readable and I was able to understand it. I have one suggestion and one question.

Suggestion: in the last example (d1 to d2) and (d1 to d3) mention `ethernet.dst` explicitly as the field with `auto` setting, so that first time readers would understand what do we use `auto` mode for.

Question: how would `mesh` work if we have multiple `tx` endpoints without support of `auto` for `ethernet.src`?

**Comment by PrasenjitAdhikary** [comment]
> 

Modified description with an example for non-auto `ethernet.src` field, with multiple `tx` endpoints.
In the example given in the description, generated device endpoint pairs are (d1 to d3) and (d2 to d3), which can be generated with either of the following configuration:
```
	mode = one_to_one,
	tx_names = [d1, d2], 
        rx_names = [d3, d3] 
```
or:
```
	mode = mesh,
	tx_names = [d1, d2],
        rx_names = [d3] 
```
We can also add this configuration part in the description example.

**Review by bortok — APPROVED** [review]
Looks good, thanks!

File: flow/endpoint.yaml [hunk 1]
```diff
@@ -62,17 +62,34 @@ components:
...
+            the actual value(s) by the implementation.
 
-            The mesh mode creates traffic between each value in tx_names to
-            every value in rx_names except in the case where the individual 
-            tx_names value is equal to rx_names value which will be ignored.
-            Identical values in tx_names or rx_names MUST raise an error.
+            The <b>one_to_one</b> mode creates traffic sub-flow(s) between each device endpoint pair in 
+            tx_names to rx_names by index.
+            The length of tx_names and rx_names MUST be the same.
+            Same device name can be repeated multiple times in tx_names or rx_names, in any order to create desired meshing detween device(s).
```
Comment (apratimmukherjee) [inline flow/endpoint.yaml:73]: Same device name ...-> The same device endpoint name ...

File: flow/endpoint.yaml [hunk 2]
```diff
@@ -62,17 +62,34 @@ components:
...
+            For 2 values in tx_names and 2 values in rx_names, generated device endpoint pairs (each pair representing a traffic sub-flow) would be 2.
```
Comment (apratimmukherjee) [inline flow/endpoint.yaml:74]: Could rephrase as '...in rx_names , 2 device endpoint pairs would be generated ( each pair representing a traffic sub-flow ) . '

File: flow/endpoint.yaml [hunk 3]
```diff
@@ -62,17 +62,34 @@ components:
...
 
-            The one_to_one mode creates traffic between each value pair in 
-            of tx_names and rx_names by index.
-            The length of tx_names and rx_name MUST be the same.
-            Identical values in tx_names or rx_names MUST raise an error.
+            The <b>mesh</b> mode creates traffic sub-flow(s) between each value in tx_names to
+            every value in rx_names, forming the device endpoint pair(s).
+            For 2 values in tx_names and 3 values in rx_names, generated device endpoint pairs would be 2x3=6. 
+
+            A generated device endpoint pair with same device name for both transmit & receive device endpoint MUST raise an error.
```
Comment (apratimmukherjee) [inline flow/endpoint.yaml:80]: with same device name -> with same device endpoint name

File: flow/endpoint.yaml [hunk 4]
```diff
@@ -62,17 +62,34 @@ components:
...
+
+            Packet fields of type <b>auto</b> would be populated with one <b>value</b> for each device endpoint pair (representing the traffic sub-flow). 
+            The value would be determined considering transmit & receive device of the sub-flow. And the sequence of the populated value(s) 
+            would be in the order of generated device endpoint pair(s).
+            If 2 device endpoint pairs are generated (based on mode, tx_names and rx_names), say (d1 to d3) and (d2 to d3), and ethernet.dst is set as <b>auto</b>, then 
+            the auto field would be replaced by the implementation with a sequnece of 2 values, [v1,v2] where
```
Comment (apratimmukherjee) [inline flow/endpoint.yaml:86]: sequence (typo )

**Review by apratimmukherjee — APPROVED** [review]
Minor typos mentioned but otherwise changes look good to me.

File: flow/endpoint.yaml (hunk 2 above)
Comment (PrasenjitAdhikary) [reply flow/endpoint.yaml]: Done.

File: flow/endpoint.yaml (hunk 4 above)
Comment (PrasenjitAdhikary) [reply flow/endpoint.yaml]: Corrected.

File: flow/endpoint.yaml (hunk 3 above)
Comment (PrasenjitAdhikary) [reply flow/endpoint.yaml]: done.

**Comment by anjan-keysight** [comment]
Looks good to me.

**Review by anjan-keysight — APPROVED** [review]

**Review by ankur-sheth — APPROVED** [review]

**Review by alakendu — APPROVED** [review]

**Pull request merged**